merlin/models/tf/inputs/embedding.py

```diff
@@ -244,6 +245,39 @@ def __init__(
         self.sequence_combiner = sequence_combiner
         self.supports_masking = True
 
+    def select_by_tag(self, tags: Union[Tags, Sequence[Tags]]) -> Optional["EmbeddingTable"]:
+        """Select features in EmbeddingTable by tags.
+
+        Since an EmbeddingTable can be a shared-embedding table, this method filters
+        the schema for features that match the tags.
+
+        If none of the features match the tags, it will return None.
+
+        Parameters
+        ----------
+        tags: Union[Tags, Sequence[Tags]]
+            A list of tags.
+
+        Returns
+        -------
+        An EmbeddingTable if the tags match. If no features match, it returns None.
+        """
+        if not isinstance(tags, collections.Sequence):
+            tags = [tags]
+
+        config = self.get_config()
+        schema = config.pop("schema", None)
+        if not schema:
+            return
+        if not isinstance(schema, Schema):
+            schema = schema_utils.tensorflow_metadata_json_to_schema(schema)
+        selected_schema = schema.select_by_tag(tags)
+        if not selected_schema:
+            return
+        config["schema"] = schema_utils.schema_to_tensorflow_metadata_json(selected_schema)
+        embedding_table = EmbeddingTable.from_config(config, table=self.table)
```

Comment:
Passing the table here is so that we are creating a shallow copy with the same layer objects being referenced. instead of cloning the layer? Or a shared embedding table might then become independently trained which wouldn't be what we want when using this?

Reply:
Yes, I intentionally did a shallow copy. Otherwise, I think we would be creating two independent tables? For example, in a two-tower, if both item and user towers were created with `select_by_tag` and the user tower used the shared embedding table, the weights would be different with a deep copy.